Hakizimana-Clement/Advizr
# [Feature]: Implement Advanced Card UI with Authentication and Bookmarking

### PR Description
This PR enhances the application by adding new features, including a bookmark function for saving user favorites, a fully functional authentication system with login and signup forms, and a responsive layout for multiple pages. 

It introduces routing for the main pages and adds form validation using React Hook Form with Zod.

### What Does This PR Do?
This PR adds advanced UI features and navigation, including:

- Bookmark functionality on advice cards for users to save their favorite quotes.
- New login, signup, dashboard, and 404 pages, with routes set up for each.
- Form validation for user authentication forms.
- A responsive design across all new and existing pages.
### Description of Tasks to Be Completed
- [x] Task 1: Add a bookmark icon to the advice card for users to save favorite advice.
- [x] Task 2: Install and configure React Router for app navigation, adding routes for the main pages.
- [x] Task 3: Create and style a responsive login page.
- [x] Task 4: Refactor form and button components for reusability across multiple pages.
- [x] Task 5: Implement form validation on login and signup pages using React Hook Form with Zod.
- [x] Task 6: Design Not Found Page.
- [x] Task 7: Build a responsive dashboard page with the following sections:
                      - _Navbar_: Includes a profile picture and a dropdown menu for logout.
                      - _Sidebar_: Includes navigation links:
                      - _Home_: Displays random advice cards, which users can bookmark.
                      - _My Favorites_: Shows saved quotes and advice with delete icons for removal.
                      - _Settings_: Allows users to update profile details, such as username and password.\
### How Should This Be Manually Tested?
1. Clone the repository and navigate into the project directory.
2. Check out the branch ft-card-ui-advanced and install dependencies with npm install.
3. Start the application locally by running npm run dev.
4. Test the following features:
            - Use the bookmark icon on advice cards on the homepage.
            - Navigate to the login, signup, and dashboard pages via the user icon or by directly typing "users/login", "users/signup", and "users/dashboard" in the browser URL.

### Any Background Context You Want to Provide?
The react-router-dom package was added to enable navigation, and react-hook-form with Zod was introduced for seamless form validation.
### Related Issues
Issue #9
